samples/places-autocomplete-addressform/src/index.ts

```diff
@@ -56,44 +60,57 @@ function initAutocomplete() {
 function fillInAddress() {
   // Get the place details from the autocomplete object.
   const place = autocomplete.getPlace();
-
-  for (const component in componentForm) {
-    (document.getElementById(component) as HTMLInputElement).value = "";
-    (document.getElementById(component) as HTMLInputElement).disabled = false;
-  }
+  let address1 = "";
+  let postcode = "";
 
   // Get each component of the address from the place details,
   // and then fill-in the corresponding field on the form.
   for (const component of place.address_components as google.maps.GeocoderAddressComponent[]) {
```

Comment:
```suggestion
  for (const component of place.address_components) {
```

While working on this, can we remove this cast and handle undefiend as @chrisjshull noted in the js version.